### PR TITLE
Initial codebase commit

### PR DESCRIPTION
**Pull request by neelam-kushwah**
Aligned Kotlin SDK with Java SDK, added Dokka plugin for Kotlin documentation, and integrated Eclipse Dash License for dependency license summaries.

### CODE REVIEW

**Review by mikaelperrin — CHANGES_REQUESTED** [review]
I would recommend removing the nullability for many of the value types, method arguments and method return types in the patch. In addition, java Optional is not really needed in kotlin and should be replaced with nullability (i.e. `Optional<String>` becomes `String?`

File: src/main/kotlin/org/eclipse/uprotocol/cloudevent/factory/CloudEventFactory.kt [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+         */
+        fun buildBaseCloudEvent(
+            id: String?, source: String?,
+            protoPayloadBytes: ByteArray?,
+            protoPayloadSchema: String?,
+            attributes: UCloudEventAttributes
+        ): CloudEventBuilder {
+            val cloudEventBuilder: CloudEventBuilder = CloudEventBuilder.v1()
+                .withId(id)
+                .withSource(URI.create(source!!)) /* Not needed:
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/cloudevent/factory/CloudEventFactory.kt:207]: I agree. Currently, I am working on removing the non-null assertion operator from all classes and will only keep it when necessary.

File: src/main/kotlin/org/eclipse/uprotocol/cloudevent/factory/CloudEventFactory.kt [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+         * @param attributes            Additional attributes such as ttl, hash and priority.
+         * @return Returns a response CloudEvent Response for the use case of RPC Response message that failed.
+         */
+        fun failedResponse(
+            applicationUriForRPC: String?,
+            serviceMethodUri: String?,
+            requestId: String?,
+            communicationStatus: Int?,
+            attributes: UCloudEventAttributes
+        ): CloudEvent? {
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/cloudevent/factory/CloudEventFactory.kt]: Addressed the review comments. Ready for further review.

File: src/main/kotlin/org/eclipse/uprotocol/cloudevent/factory/CloudEventFactory.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/cloudevent/factory/CloudEventFactory.kt]: Addressed the review comments. Ready for further review.

**Review by stevenhartley — APPROVED** [review]

**Comment by stevenhartley** [comment]
Merging the initial release so that we can workout the workflows and last minute 1.5 changes.

**Review by mikaelperrin — COMMENTED** [review]
Although this PR is closed, there is still some java-like code that we may want to convert to kotlin.

File: src/main/kotlin/org/eclipse/uprotocol/rpc/RpcMapper.kt [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    companion object {
+        /**
+         * Map a response of CompletableFuture&lt;UPayload&gt; from Link into a CompletableFuture containing the declared expected return type of the RPC method or an exception.
+         * @param responseFuture CompletableFuture&lt;UPayload&gt; response from uTransport.
+         * @param expectedClazz The class name of the declared expected return type of the RPC method.
+         * @return Returns a CompletableFuture containing the declared expected return type of the RPC method or an exception.
+         * @param <T> The declared expected return type of the RPC method.
+        </T> */
+        fun <T : Message> mapResponse(
+            responseFuture: CompletableFuture<UPayload>,
```
Comment (mikaelperrin) [inline src/main/kotlin/org/eclipse/uprotocol/rpc/RpcMapper.kt:51]: should we consider coroutines here (i.e. kotlin flow or suspendable functions)? CompletableFuture is the java async type.

**Review by mikaelperrin — COMMENTED** [review]
CompletableFuture usage is an issue here. We should think of leveraging async features from kotlin instead.

File: src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UPayload.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    override fun hashCode(): Int {
+        return Objects.hash(data.contentHashCode(), hint)
+    }
+
+    override fun toString(): String {
+        return "UPayload{" +
+                "data=" + data().contentToString() +
+                ", hint=" + hint + '}'
+    }
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UPayload.kt:81]: After converting the class to a data class, a warning is generated due to the presence of a property with an 'Array' type. It is advised to override the 'equals()' and 'hashCode()' methods to ensure proper functionality when comparing instances of the data class.

File: src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UStatus.kt [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            return value
+        }
+
+        companion object {
+            /**
+             * Get the Code from an integer value.
+             * @param value The integer value of the Code.
+             * @return Returns the Code if found, otherwise returns Optional.empty().
+             */
+            fun from(value: Int): Optional<Code> {
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UStatus.kt:58]: If we replace "optional" with "nullable," the property will return either a value or null. However, it's important to note that we don't want to return null. To confirm whether this change is appropriate or not, I suggest @stevenhartley provide input here

File: src/main/kotlin/org/eclipse/uprotocol/transport/builder/UAttributesBuilder.kt [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    companion object {
+        /**
+         * Construct a UAttributesBuilder for a publish message.
+         * @param priority The priority of the message.
+         * @return Returns the UAttributesBuilder with the configured priority.
+         */
+        fun publish(priority: UPriority): UAttributesBuilder {
+            Objects.requireNonNull(priority, "UPriority cannot be null.")
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/transport/builder/UAttributesBuilder.kt:151]: Addressed comments; ready for further review. New pull request created.

File: src/main/kotlin/org/eclipse/uprotocol/transport/builder/UAttributesBuilder.kt [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+         * Construct a UAttributesBuilder for a request message.
+         * @param priority The priority of the message.
+         * @param sink The destination URI.
+         * @param ttl The time to live in milliseconds.
+         * @return Returns the UAttributesBuilder with the configured priority, sink and ttl.
+         */
+        fun request(priority: UPriority, sink: UUri, ttl: Int): UAttributesBuilder {
+            Objects.requireNonNull(priority, "UPriority cannot be null.")
+            Objects.requireNonNull(ttl, "ttl cannot be null.")
+            Objects.requireNonNull(sink, "sink cannot be null.")
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/transport/builder/UAttributesBuilder.kt:179]: Addressed comments; ready for further review. New pull request created.

File: src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UPayload.kt [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+ * The UPayload contains the clean Payload information along with its raw serialized structure of a byte[].
+ */
+class UPayload(data: ByteArray?, hint: UPayloadFormat?) {
+    private val data: ByteArray
+    private val hint // Hint regarding the bytes contained within the UPayload
+            : UPayloadFormat
+
+
+    init {
+        this.data = Objects.requireNonNullElse(data, ByteArray(0))
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UPayload.kt:40]: Addressed comments; ready for further review. New pull request created.

File: src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UStatus.kt [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+                return Arrays.stream(entries.toTypedArray()).filter { p: Code -> p.value() == value }.findAny()
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UStatus.kt:59]: Addressed comments; ready for further review. New pull request created.

File: src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UPayload.kt [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        return data
+    }
+
+    /**
+     * The hint regarding the bytes contained within the UPayload.
+     * @return Returns the hint regarding the bytes contained within the UPayload.
+     */
+    fun hint(): UPayloadFormat {
+        return hint
+    }
```
[the remark this answers is not in the document]
Comment (neelam-kushwah) [reply src/main/kotlin/org/eclipse/uprotocol/transport/datamodel/UPayload.kt:58]: Addressed comments; ready for further review. New pull request created.

**Pull request merged**